MekHQ/src/mekhq/campaign/universe/Planet.java

```diff
@@ -314,18 +297,22 @@ public Planet(String tsvData, List<DateTime> years) throws Exception {
 
                 // dirty hack here assumes that there's only one faction per event, which is true in the case
                 // of this spreadsheet
-                if(x == 3 || !eventList.get(eventList.size() - 1).faction.get(0).equals(newFaction)) {
-                    PlanetaryEvent pe = new PlanetaryEvent();
-                    DateTime eventDate = years.get(x - 3);
-                    pe.faction = new ArrayList<String>();
+                if ((x == 3) || !eventList.get(eventList.size() - 1).faction.get(0).equals(newFaction)) {
+                    LocalDate eventDate = years.get(x - 3);
+
+                    PlanetaryEvent pe = events.getOrDefault(eventDate, new PlanetaryEvent());
+                    pe.faction = new ArrayList<>();
                     pe.faction.add(newFaction);
-                    pe.date = eventDate;
 
-                    this.eventList.add(pe);
-                    this.events.put(eventDate, pe);
+                    if (!events.containsKey(eventDate)) {
```

Comment:
This seems like it's a behavior change from existing behavior, can you go into a little detail on why it was done?

Reply:
This was done to improve the code in question (as a single event can change anything on a given day) by removing the same day event hack of placing an event slightly later in the day, which is no longer possible (LocalDate only stores year, month, day). It was also significantly easier to do this than replace it with LocalDateTime, which would require conversions.

Reply:
Sounds good in theory, although in practice, we'll need to take a look at the planetary events data and see if we have any events like that which will need to be merged (in order to avoid throwing away any events past the first).

Reply:
That is already handled in the code, this is just a place that was missed.

Reply:
Although the changes should use that method, so I'll fit that bit.

Reply:
And the changes are fixed, it properly adds it to the eventList now without causing issues from duplicate events (which currently will occur).